plugin/ipythonPlugins/src/dist/iruby/iruby.js

```diff
@@ -53,13 +53,22 @@ define(function(require, exports, module) {
           shellID = myPython.utils.uuid();
         }
 
-        var base = _.string.startsWith(serviceBase, "/") ? serviceBase : "/" + serviceBase;
-        bkHelper.httpGet("../beaker/rest/plugin-services/getIPythonPassword", {pluginId: PLUGIN_NAME})
+        bkHelper.httpGet(bkHelper.serverUrl("beaker/rest/plugin-services/getIPythonPassword"), {pluginId: PLUGIN_NAME})
         .success(function(result) {
-          bkHelper.httpPost(base + "/login?next=%2E", {password: result})
+          bkHelper.httpPost(bkHelper.serverUrl(serviceBase + "/login?next=%2E"), {password: result})
           .success(function(result) {
+            var baseurl = bkHelper.serverUrl(serviceBase);
```

Comment:
Perhaps I do not follow, I meant this logic

```js
+            var t = baseurl.indexOf('//');
+            if (t>=0) {
+              baseurl = baseurl.substring(t+2);
+              t = baseurl.indexOf('/');
+              if (t>=0) {
+                baseurl = baseurl.substring(t);
+              }
+            }
```

Could be abstracted to a shared named function on bkHelper (or something) so that the logic is not copy pasted across each plugin.

What do you think?